### PR TITLE
Webcam recorder

### PR DESCRIPTION
**Pull request by AdrienMazet**

### CODE REVIEW

File: packages/front/src/components/recorder/WebcamRecorder.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  const mediaRecorderRef: MutableRefObject<MediaRecorder | null> = useRef(null);
+
+  const [videoUrl, setVideoUrl] = useState("");
+  const [error, setError] = useState("");
+  const [state, setState] = useState(RecorderState.LOADING);
+
+  const handleStartRecording = () => {
+    setState(RecorderState.RECORDING);
+    mediaRecorderRef.current = new MediaRecorder(
+      webcamRef.current?.stream as MediaStream,
```
Comment (rigwild) [inline packages/front/src/components/recorder/WebcamRecorder.tsx:36]: Is it transpiled to an older version of JavaScript using something like Babel ? [Optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) is supported from Chrome 80, which is too new.

Comment (AdrienMazet) [reply packages/front/src/components/recorder/WebcamRecorder.tsx:36]: Transpiled to the es2019 standard as specified in the tsconfig.json. Create-react-app use a babel preset in top of that.

**Review by rigwild — APPROVED** [review]
lgtm.
I think it would be nice to have some component that makes it more obvious that it is recording.
It could be a sound/voice visualizer or just a red "record" animated icon.

**Comment by AdrienMazet** [comment]
> lgtm.
> I think it would be nice to have some component that makes it more obvious that it is recording.
> It could be a sound/voice visualizer or just a red "record" animated icon.

https://ant.design/components/icon/ : VideoCameraTwoTone (just for reminder)

I can use this with a text : "recording..."

**Comment by AdrienMazet** [comment]
> lgtm.
> I think it would be nice to have some component that makes it more obvious that it is recording.
> It could be a sound/voice visualizer or just a red "record" animated icon.

Done : 83a373b38ea073401e92b60a7c829d96758dd458

**Review by rigwild — APPROVED** [review]

**Pull request merged**